core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -822,62 +822,24 @@ Cuda::size_type Cuda::device_arch() {
 void Cuda::impl_finalize() { Impl::CudaInternal::singleton().finalize(); }
 
 Cuda::Cuda()
-    : m_space_instance(&Impl::CudaInternal::singleton()), m_counter(nullptr) {
+    : m_space_instance(Kokkos::Experimental::UnmanagedPtr<Impl::CudaInternal>(
+          &Impl::CudaInternal::singleton())) {
```

Comment:
I think this is more complicated than it needs to be. We do NOT need to make copies of the singleton instance different than any other one in my opinion.

Reply:
I mean, this essentially preserves performance for the default case, assuming reasonable branch prediction. Are we really confident that no one is copying instances of `Cuda` in a performance-sensitive context? If so, why do we need `UnmanagedPtr` at all?

Reply:
For View I thought?